1/contrib/s2i/run

```diff
@@ -12,6 +12,24 @@ source /usr/local/bin/kube-slave-common.sh
 jenkins_config_dir="/opt/openshift/configuration"
 jenkins_config_path="${jenkins_config_dir}/config.xml"
 
+
+CONTAINER_MEMORY_IN_BYTES=`cat /sys/fs/cgroup/memory/memory.limit_in_bytes`
+echo "Container memory in bytes ${CONTAINER_MEMORY_IN_BYTES}"
+DEFAULT_MEMORY_CEILING=$((2**30-1))
+echo "Default memory ceiling ${DEFAULT_MEMORY_CEILING}"
+if [ "${CONTAINER_MEMORY_IN_BYTES}" -lt "${DEFAULT_MEMORY_CEILING}" ]; then
+    if [ -z $CONTAINER_HEAP_PERCENT ]; then
+	CONTAINER_HEAP_PERCENT=0.80
+    fi
+    
+    CONTAINER_MEMORY_IN_MB=$((${CONTAINER_MEMORY_IN_BYTES}/1024**2))
+    CONTAINER_HEAP_MAX=$(echo "${CONTAINER_MEMORY_IN_MB} ${CONTAINER_HEAP_PERCENT}" | awk '{ printf "%d", $1 * $2 }')
+    
+    echo "Going to set max heap to ${CONTAINER_HEAP_MAX}"
```

Comment:
i don't think this needs to be printed either.  you can always rsh into the pod and run ps if you want to see what the java args were.  If you do want to print it (eg in case it's needed for debugging a non-starting container), make the message something more formal like "Java max heap=$FOO"